apps/arweave/src/ar_nonce_limiter.erl

```diff
@@ -118,6 +127,17 @@ get_step_checkpoints(StepNumber, NextSeed, StartIntervalNumber, NextVDFDifficult
 get_step_checkpoints(StepNumber, SessionKey) ->
 	gen_server:call(?MODULE, {get_step_checkpoints, StepNumber, SessionKey}, infinity).
 
+%% @doc Return the cached last step checkpoints, seed, and partition upper bound
+%% for the given step.
+%% Return not_found if the VDF session or cached last step checkpoints are not found.
+get_step_checkpoints_seed_upper_bound(StepNumber, NextSeed, StartIntervalNumber,
```

Comment:
As a way to better understand how all this works together I'm trying to wrap my head around the need to have both `get_step_checkpoints_seed_upper_bound` and `get_triplets`... is there any way to merge the two? Or name them something that clarifies how they're used?

`get_step_checkpoints_seed_upper_bound` in particular is a mouthful and I find myself getting lost as I try to keep everything in my head as I trace through the code. Could we split it and use `get_step_checkpoints` as is? And add an additional call or calls to query the seed and upper bound?

Reply:
`get_step_checkpoints_seed_upper_bound` fetches the information about a particular step while `get_triplets` returns the latest range.

What do you think about `get_step_info(StepN, SessionKey, [last_step_checkpoints, seed, partition_upper_bound])` where the last argument is a list of quantities that we'd like to receive?

Reply:
I like it. It unifies the naming and makes clear what the function is returning.

Reply:
On second thought, get_step_info introduces a bit of unpleasant complexity, I broke down the function into three instead - https://github.com/ArweaveTeam/arweave/pull/515/commits/6dffaa2ca2204bcc0392b42b5b39959c162ccc98